src/biophysics.jl

```diff
@@ -284,8 +284,35 @@ function convection(Body, area, T_air, T_surf, vel, P_atmos, elev, fluid)
     (Q_conv=Q_conv, Hc=Hc, Hd=Hd, Sh=Sh, Q_free=Q_free, Hc_free=Hc_free, Hc_forc=Hc_forc, Sh_free=Sh_free, Sh_forc=Sh_forc, Hd_free=Hd_free, Hd_forc=Hd_forc)
 end
 
-function conduction(A, L, T_org, T_sub, k_sub)
-    A * (k_sub / L) * (T_org - T_sub)
+convection(Body, p::Model, o::OrganismalVars, e::EnvironmentalVars) = begin
+    model_pars = stripparams(p)
+    env_pars = model_pars[2]
+    Ac = Body.geometry.area * (1 - o.p_cond)
+    convection(Body, Ac, e.Ta, o.T_surf, e.vel, env_pars.P_atmos, env_pars.elev, env_pars.fluid)
+end
+
+convection(T_x, Body, p::Model, o::OrganismalVars, e::EnvironmentalVars) = begin
+    model_pars = stripparams(p)
+    env_pars = model_pars[2]
+    Ac = Body.geometry.area * (1 - o.p_cond)
+    convection(Body, Ac, e.Ta, T_x, e.vel, env_pars.P_atmos, env_pars.elev, env_pars.fluid)
+end
+
+
+function conduction(A, L, T_surf, T_sub, k_sub)
+    A * (k_sub / L) * (T_surf - T_sub)
+end
+
+conduction(Body, p::Model, o::OrganismalVars, e::EnvironmentalVars) = begin
```

Comment:
I wouldn't pass a `Model` around like this... better if you unwrap it in the furthest out function and separate that from any real code.

Our own functions should dispatch on our own model type, after `stripparams` has run. Then we can define lots of different kinds of model rather than one. The model wrapper is for user handling in the REPL.

Also wondering what is the difference between whats inside the Model and Body? I would usually lump those into one object that goes inside `Model`, and gets unwrapped for dispatch.

Reply:
So you mean constructing a Model like this:

model = Model((body=body_organism, 
                parameters = (OrganismalPars(), EnvironmentalPars()), 
                variables = (OrganismalVars(), EnvironmentalVars())))
model = stripparams(model_params)

or something like this?

Reply:
I don't completely understand. You would write a function that takes the `Model` object, e.g.:
```
heat_balance(mod::Model) = begin
    model_pars = stripparams(mod)
    heat_balance(model_pars, Naked())
end
```
And then:
```
heat_balance(model_pars, ::Naked) = begin
.......
end
```
But I still have to unpack parameters and variables... even if I run `stripparams` params are not available but wrapped in a `Tuple`.

I am missing something (or many things...)

Reply:
Just a few things!

I don't mean to lump variables with parameters, normally they would be separate. But e.g. `Naked` would be a part of the model a user would defined.

So you would have an `Organism` that has a body and a `body` method:

```julia
# We have a generic abstract type for all organisms
abstract type Organism end

# Then define a Lizard
struct Lizard{B<:Body,P<:OrganismParams} <: Organism
    body::B
    params::P
end
body(o::Lizard) = l.body
params(o::Lizard) = l.params

# Then the heat balance function is, for example (I don't know exactly what it needs to be):
# A method dispatching on a `Model`
heat_balance(mod::Model, vars) = heat_balance(stripparams(mod), vars)
# A generic method that expands dispatch to include the body type and param (an example, if we need that)
heat_balance(organism, vars) = heat_balance(body(organism), params(organism), organism, var)
# Some generic `Body` method
function heat_balance(body::Body, params, organism, vars) 
    # Do the math for a generic organism, with current vars
end
# Some specific method for DesertIguana (not saying we need this, just an example of what is possible)
function heat_balance(body::DesertIguana, params, organism, vars) 
    # Do the math for a DesertIguana, with current vars
end
```

And this will work either with the raw struct or if it is wrapped in `Model`.

If we needs to be more flexible and scripty, lizard can just a be `NamedTuple`

```julia
lizard = (
    body = Body(Named(), LizardGeom()),
    params = LizardParams()
)
heat_balance(lizard, vars) 
```

And we could define a method to handle getting the fields of the NamedTuple:
```julia
heat_balance(o::NamedTuple, vars) = heat_balance(o.body, o.params, o, vars)
```

Or instead a generic fallback for `body` and `params` methods to get the fields of the same names from any object:
```julia
body(x) = getproperty(x, :body)
params(x) = getproperty(x, :params)
```

Reply:
That. Is. Brilliant. Thanks, Raf! That's the flavor of julia that I'm still missing. 

I will try to implement this in this branch tonight :)

Reply:
I think I mixed up shape and insulation but you get the idea!

We may need to add some more methods to e.g. `Body` to be able to get insulation and shape from it so these things actually work.